File: frameowkrs/typescript-playwright/pages/playground_page.ts
import { expect, type Page } from '@playwright/test';

export class PlaygroundPage {
  constructor(private readonly page: Page) {}

  async openWithSeed(seed: number): Promise<void> {
    await this.page.addInitScript((value) => localStorage.setItem('flaky_seed', String(value)), seed);
    await this.page.goto('/playground', { waitUntil: 'commit' });
    await expect(this.page.getByTestId('flaky-seed-display')).toContainText(String(seed));
  }

  async runFastSuccess(seed: number): Promise<void> {
    await this.page.getByTestId('trigger-success-fast').click();
    await expect(this.page.getByTestId('flaky-invoice-modal')).toBeVisible({ timeout: 8_000 });
    await expect(this.page.getByTestId('flaky-invoice-status')).toBeVisible();
    await expect(this.page.getByTestId('flaky-seed-display')).toContainText(String(seed));
  }
}

File: frameowkrs/python-playwright/pages/playground_page.py
import json

from playwright.sync_api import Page, expect


class PlaygroundPage:
    def __init__(self, page: Page) -> None:
        self.page = page

    def open_with_seed(self, seed: int) -> None:
        self.page.add_init_script(
            f"localStorage.setItem('flaky_seed', {json.dumps(str(seed))})"
        )
        self.page.goto("/playground")
        expect(self.page.get_by_test_id("flaky-seed-display")).to_contain_text(str(seed))

    def run_fast_success(self) -> None:
        self.page.get_by_test_id("trigger-success-fast").click()
        expect(self.page.get_by_test_id("flaky-invoice-modal")).to_be_visible(timeout=8000)
        expect(self.page.get_by_test_id("flaky-invoice-status")).to_be_visible()
